czkawka_core/src/big_file.rs

```diff
@@ -144,59 +143,40 @@ impl BigFile {
-                        + "/";
-
-                    for ed in &self.directories.excluded_directories {
-                        if next_folder == *ed {
-                            continue 'dir;
-                        }
+                    let next_folder = current_folder.join(entry_data.file_name());
+                    if self.directories.excluded_directories.contains(&next_folder) {
+                        continue 'dir;
                     }
+
                     for expression in &self.excluded_items.items {
                         if Common::regex_check(expression, &next_folder) {
                             continue 'dir;
                         }
                     }
                     folders_to_check.push(next_folder);
                 } else if metadata.is_file() {
-                    let file_name_lowercase: String = match entry_data.file_name().into_string() {
-                        Ok(t) => t,
-                        Err(_) => continue,
-                    }
-                    .to_lowercase();
+                    // Extracting file extension
+                    let file_extension = entry_data.path().extension().and_then(OsStr::to_str).map(str::to_lowercase);
 
                     // Checking allowed extensions
                     if !self.allowed_extensions.file_extensions.is_empty() {
-                        let allowed = self.allowed_extensions.file_extensions.iter().any(|e| file_name_lowercase.ends_with((".".to_string() + e.to_lowercase().as_str()).as_str()));
+                        let allowed = self.allowed_extensions.file_extensions.iter().map(|e| e.to_lowercase()).any(|e| file_extension == Some(e));
                         if !allowed {
                             // Not an allowed extension, ignore it.
                             self.information.number_of_ignored_files += 1;
                             continue 'dir;
                         }
                     }
-                    // Checking files
-                    #[allow(unused_mut)] // Used is later by Windows build
-                    let mut current_file_name = "".to_owned()
-                        + &current_folder
-                        + match &entry_data.file_name().into_string() {
-                            Ok(t) => t,
-                            Err(_) => continue,
-                        };
 
                     // Checking expressions
+                    let mut current_file_name = current_folder.join(entry_data.file_name());
                     for expression in &self.excluded_items.items {
                         if Common::regex_check(expression, &current_file_name) {
                             continue 'dir;
                         }
                     }
 
-                    #[cfg(target_family = "windows")]
-                    {
+                    if cfg!(target_family = "windows") {
                         current_file_name = Common::prettier_windows_path(&current_file_name);
```

Comment:
`prettier_windows_path` function was used only for normalizing path like `c:/Av`, `c:\AV`, `C:/av` to `C:/av` to allow easily remove duplicate path or overlap ones.
With PathBuf I don't think that this is necessary, and can be removed(of course if everything else will work fine)